bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/ImageImpl.java

```diff
@@ -238,6 +244,42 @@ protected void initModel() {
         }
     }
 
+    /**
+     * Extracts the asset name from DAM resource
+     * 
+     * @return image name from DAM
+     */
+    protected String getImageNameFromDam() {
+        String imageName = "";
+        Resource damResource = request.getResourceResolver().getResource(fileReference);
+        if (damResource != null) {
+          Asset asset = damResource.adaptTo(Asset.class);
+            imageName = asset != null ? StringUtils.trimToNull(asset.getName()) : "";
+        }
+        return getSeoFriendlyName(FilenameUtils.getBaseName(imageName));
+    }
+    
+    /**
+     * Content editors can store DAM assets with white spaces in the name, this method makes
+     * the asset name SEO friendly, Translates the string into {@code application/x-www-form-urlencoded} 
+     * format using {@code utf-8} encoding scheme.
+     * 
+     * @param assetName
+     * @return name of the asset without extension
+     */
+    protected String getSeoFriendlyName(String imageName) {
+      
+      //Google recommends to use hyphens (-) instead of underscores (_) for seo. See https://support.google.com/webmasters/answer/76329?hl=en 
+      String seoFriendlyName = imageName.replaceAll("[\\ _]", "-").toLowerCase();
+        try {
```

Comment:
Formatting, indentation looks a little off here as well (we should make sure we're using spaces not tabs).

Reply:
I have formatted with 4 spaces(Space only)